src/docgen/transpile/java.ts

```diff
@@ -0,0 +1,356 @@
+      for (const property of struct.allProperties) {
+        const parameter = propertyToParameter(callable, property);
+        parameters.push(parameter);
+      }
+    }
+
+    return {
+      name,
+      parentType: type,
+      import: this.formatImport(type),
+      parameters,
+      signatures,
+      invocations,
+    };
+  }
+
+  public class(klass: reflect.ClassType): transpile.TranspiledClass {
+    return {
+      name: klass.name,
+      type: this.type(klass),
+    };
+  }
+
+  public struct(struct: reflect.InterfaceType): transpile.TranspiledStruct {
+    const type = this.type(struct);
+    const inputs = struct.allProperties.map((p) =>
+      this.formatBuilderMethod(this.property(p)),
+    ).flat();
+    return {
+      type: type,
+      name: struct.name,
+      import: this.formatImport(type),
+      initialization: this.formatStructBuilder(type, inputs),
+    };
+  }
+
+  public interface(
+    iface: reflect.InterfaceType,
+  ): transpile.TranspiledInterface {
+    return {
+      name: iface.name,
+      type: this.type(iface),
+    };
+  }
+
+  public parameter(
+    parameter: reflect.Parameter,
+  ): transpile.TranspiledParameter {
+    return {
+      name: parameter.name,
+      parentType: this.type(parameter.parentType),
+      typeReference: this.typeReference(parameter.type),
+      optional: parameter.optional,
+    };
+  }
+
+  public property(property: reflect.Property): transpile.TranspiledProperty {
+    return {
+      name: property.name,
+      parentType: this.type(property.parentType),
+      typeReference: this.typeReference(property.type),
+      optional: property.optional,
+    };
+  }
+
+  public enum(enu: reflect.EnumType): transpile.TranspiledEnum {
+    return {
+      fqn: this.type(enu).fqn,
+      name: enu.name,
+    };
+  }
+
+  public enumMember(em: reflect.EnumMember): transpile.TranspiledEnumMember {
+    return {
+      fqn: `${this.enum(em.enumType).fqn}.${em.name}`,
+      name: em.name,
+    };
+  }
+
+  public unionOf(types: string[]): string {
+    return types.join(' OR ');
+  }
+
+  public listOf(type: string): string {
+    return `java.util.List<${type}>`;
+  }
+
+  public mapOf(type: string): string {
+    return `java.util.Map<java.lang.String, ${type}>`;
+  }
+
+  public any(): string {
+    return 'java.lang.Object';
+  }
+
+  public str(): string {
+    return 'java.lang.String';
+  }
+
+  public number(): string {
+    return 'java.lang.Number';
+  }
+
+  public boolean(): string {
+    return 'java.lang.Boolean';
+  }
+
+  public json(): string {
+    return 'com.fasterxml.jackson.databind.node.ObjectNode';
+  }
+
+  public date(): string {
+    return 'java.time.Instant';
+  }
+
+  public readme(readme: string): string {
+    return readme;
+  }
+
+  private formatImport(type: transpile.TranspiledType): string {
+    const namespace = type.namespace ? `.${type.namespace}` : '';
+    return `import ${type.module}${namespace}.${type.name};`;
```

Comment:
I wonder - does this work?

```suggestion
    return `import ${type.fqn}`;
```

Reply:
You're right, good catch!